cmd/vic-machine/common/completion.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/urfave/cli.v1"
+)
+
+// getNames extracts one or more names from a cli.Flag object, trimming whitespace from each.
+func getNames(flag cli.Flag) []string {
+	names := strings.Split(flag.GetName(), ",")
+
+	var trimmedNames []string
+	for _, name := range names {
+		trimmedNames = append(trimmedNames, strings.Trim(name, " "))
+	}
+
+	return trimmedNames
+}
+
+// findFlag searches a haystack of cli.Flag objects for the first with a name matching the needle.
+func findFlag(needle string, haystack []cli.Flag) cli.Flag {
+	for _, f := range haystack {
```

Comment:
A couple of factors led me here:

1. To do better, I need a sorted list. With the sorted list, I could do searches in `O(log(n*m))`. However initializing that sorted list actually takes `O(n*m*log(n*m))`. If I needed to _find_ a lot of things (more than `log(n*m)`), that might be worth it, but in this case I'm only ever searching once so the initialization cost would dominate.
2. In our code, `getNames` always returns 1-2 elements (a long flag and maybe a short flag), so that's essential a constant factor.
3. The current pattern allows us to bail out as soon as we've found what we're looking for.

Reply:
Yah, that's why I said it works fine. Even if there were hundreds of elements in each list, it would not be really noticeable.